aries-test-harness/features/0037-present-proof.feature

```diff
@@ -0,0 +1,94 @@
+   Scenario Outline: Present Proof of specific types and proof is acknowledged
+      Given "2" agents
+         | name  | role     |
+         | Faber | verifier |
+         | Bob   | prover   |
+      And "Faber" and "Bob" have an existing connection
+      And "Bob" has an issued credential from <issuer>
+      When "Faber" sends a request for proof presentation to "Bob" for <proof>
+      And "Bob" makes the presentation of the proof
+      And "Faber" acknowledges the proof
+      Then "Bob" has the proof acknowledged
+
+      # Fyi, some of these proof examples may end up in the negative/exception test case. leaving them here for now.
+      # This proof list is free to grow as cases are discovered and raised.
+      Examples:
+         | issuer | proof       |
+         | Acme   | address2    |
+         | Faber  | zip         |
+         |        | city        |
+         |        | address1    |
+         |        | state       |
+         |        | empty       |
+         |        | null        |
+         |        | true        |
+         |        | false       |
+         |        | max i32     |
+         |        | max i32 + 1 |
+         |        | i0          |
+         |        | min i32     |
+         |        | min i32 - 1 |
+         |        | float 0.0   |
+         |        | str 0.0     |
+         |        | chr 0       |
+         |        | chr 1       |
+         |        | chr 2       |
+
+   @T002-API10-RFC0037 @P1 @AcceptanceTest @wip @NeedsReview
+   Scenario Outline: Present Proof where the prover has proposed the presentation of proof and is acknowledged
+      Given "2" agents
+         | name  | role     |
+         | Faber | verifier |
+         | Bob   | prover   |
+      And "Faber" and "Bob" have an existing connection
+      And "Bob" has an issued credential from <issuer>
+      When "Faber" sends a request for proof presentation to "Bob"
+      And "Bob" doesn’t want to reveal what was requested so makes a presentation proposal
+      And "Faber" agrees to continue so sends a request for proof presentation
+      And "Bob" makes the presentation of the proof
+      And "Faber" acknowledges the proof
+      Then "Bob" has the proof acknowledged
+
+      Examples:
+         | issuer |
+         | Acme   |
+         | Faber  |
+
+   @T003-API10-RFC0037 @AcceptanceTest @ExceptionTest @P2 @wip @NeedsReview
+   Scenario Outline: Present Proof where the verifier rejects the presentation of the proof
+      Given "2" agents
+         | name  | role     |
+         | Faber | verifier |
+         | Bob   | prover   |
+      And "Faber" and "Bob" have an existing connection
+      And "Bob" has an issued credential from <issuer>
+      When "Faber" sends a request for proof presentation to "Bob"
+      And "Bob" makes the presentation of the proof
+      And "Faber" rejects the proof so sends a presentation rejection
+      Then "Bob" has the proof unacknowledged
```

Comment:
Worth a separate closing state to note this was `rejected` and is not still pending, for example?

Reply:
That is a good question. There are a set of states in the 1.0 version of Present Proof, but nothing that indicates that that closing unacknowledged state actually is. I'm going to assume the state is null? We can verify once we get that step executing. Then raise it as a bug or go forward with it as it is. The 2.0 version has an "abandoned" state which I think is what it would be set to in this case. What ever that ending state is, i think it should be asserted in that last step. This way the same Gherkin can be used for 1.0 and 2.0 and only the state check would be different. I'm also hesitant on using/checking states in 1.0 version of the tests because the RFC makes no mention of what the states are. I had to reach into the 1.0 protocol code to find out.